Sources/Locator.swift

```diff
@@ -67,11 +67,10 @@ public class LocatorManager: NSObject, CLLocationManagerDelegate {
 		/// - Parameter callback: callback to call
 		/// - Returns: token used to remove the listener in a second time.
 		public func listen(forAuthChanges callback: @escaping AuthorizationDidChangeEvent) -> Token {
-			var (next,overflow) = self.nextTokenID.addingReportingOverflow(1)
-			if overflow {
-				next = 0
-			}
-			self.callbacks[next] = callback
+            let isNotOverflowed = nextTokenID < UInt64.max
```

Comment:
@AgranatMarkit why did you swap `addingReportingOverflow` function call for an `isNotOverflowed` flag? It makes `listen` function much less safe because adding a number greater than 1 to `nextTokenID` would result in an overflow.

Reply:
Good Point.  
I did it for these concrete situation, when we need obtain nextTokenID by adding 1.
The key difference between my implementation and old is that when overflow occurs i rewrite old callbacks with new values. In old implementation, if we face overflow we will not have ablility to add new listeners. As i mention the drawback of my approach is that we will lost old subscription by rewriting it with new closures.  
To avoid this problem, we need change Type of token to String, that will allow to avoid overflow problems. But it will break current interface and will lead to increase size of Token in memory. It would be nice to hear opinion of another contributors.

Reply:
i return to old version of incrementing. But at current context it's no matter how we check overflow.